modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/RichText/RichText.es.js

```diff
@@ -12,96 +12,57 @@
  * details.
  */
 
-import {Editor} from 'frontend-editor-ckeditor-web';
-import React from 'react';
+import {ClassicEditor} from 'frontend-editor-ckeditor-web';
+import React, {useMemo} from 'react';
 
 import {FieldBase} from '../FieldBase/ReactFieldBase.es';
 import {useSyncValue} from '../hooks/useSyncValue.es';
 
 const CKEDITOR_CONFIG = {
-	toolbar: [
-		{items: ['Undo', 'Redo'], name: 'clipboard'},
-		'/',
-		{
-			items: [
-				'Bold',
-				'Italic',
-				'Underline',
-				'Strike',
-				'-',
-				'CopyFormatting',
-				'RemoveFormat',
-			],
-			name: 'basicstyles',
-		},
-		{
-			items: [
-				'NumberedList',
-				'BulletedList',
-				'-',
-				'Outdent',
-				'Indent',
-				'-',
-				'Blockquote',
-				'-',
-				'JustifyLeft',
-				'JustifyCenter',
-				'JustifyRight',
-				'JustifyBlock',
-			],
-			name: 'paragraph',
-		},
-		{items: ['Link', 'Unlink', 'Anchor'], name: 'links'},
-		{
-			items: ['Image', 'Table', 'HorizontalRule', 'SpecialChar'],
-			name: 'insert',
-		},
-		'/',
-		{items: ['Styles', 'Format', 'Font', 'FontSize'], name: 'styles'},
-		{items: ['TextColor', 'BGColor'], name: 'colors'},
-		{items: ['Maximize'], name: 'tools'},
-		{
-			items: ['Source'],
-			name: 'document',
-		},
-	],
+	autoGrow_bottomSpace: 50,
+	autoGrow_maxHeight: 600,
+	autoGrow_minHeight: 215,
+	resize_enabled: true,
 };
 
 const RichText = ({
+	editorConfig,
 	id,
 	name,
 	onChange,
 	predefinedValue,
 	readOnly,
 	value,
+	visible,
 	...otherProps
 }) => {
 	const [currentValue, setCurrentValue] = useSyncValue(
 		value ? value : predefinedValue
 	);
 
-	const editorProps = {
-		config: CKEDITOR_CONFIG,
-		data: currentValue,
-	};
-
-	if (readOnly) {
-		editorProps.readOnly = true;
-		editorProps.style = {pointerEvents: 'none'};
-	}
-	else {
-		editorProps.onChange = (event) => {
-			const newValue = event.editor.getData();
-
-			setCurrentValue(newValue);
-
-			onChange(event, newValue);
-		};
-	}
-
 	return (
-		<FieldBase {...otherProps} id={id} name={name} readOnly={readOnly}>
-			<Editor {...editorProps} />
+		<FieldBase
+			{...otherProps}
+			id={id}
+			name={name}
+			readOnly={readOnly}
+			style={readOnly ? {pointerEvents: 'none'} : null}
+			visible={visible}
+		>
```

Comment:
One comment here... do we really need to instantiate the `CKEditor` in edition mode? The editor is already heavy enough without a readOnly instance of CKEditor in there.

In the case of "Geolocation", we [render a simple image](https://github.com/liferay-frontend/liferay-portal/blob/master/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Geolocation/Geolocation.es.js#L75-L79) when we're in `viewMode`. I feel like something similar would be benefitial here for the sake of performance.

Reply:
@jbalsas I'm not sure if we could do this, because it will be on the Builder's Preview side, I think Design wants the user to be able to see the same fields being rendered.

Reply:
I know, that's why we implemented this this way... but they already have exceptions for this rule, and I think it serves very little rendering the whole thing for nothing.

Could you run it by them, just to see if this is a must or is something that can be tweaked?

Reply:
Sure, I agree. I'll check with them today at the daily.